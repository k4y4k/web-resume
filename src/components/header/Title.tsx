import * as React from 'react'

interface TitleProps {
  title: string
}

const Title = ({ title }: TitleProps): JSX.Element | null => {
  if (title === '') return null

  return <h1 data-testid='headerTitle'>{title}</h1>
}

export default Title
