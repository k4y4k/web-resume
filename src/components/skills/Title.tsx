import * as React from 'react'

interface TitleTypes {
  category: string
}

const Title = ({ category }: TitleTypes): JSX.Element | null => {
  if (category === '') return null

  return <h2 data-testid='skillsTitle'>{category}</h2>
}

export default Title
