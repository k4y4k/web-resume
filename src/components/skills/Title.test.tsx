import * as React from "react";
import { render, screen } from "@testing-library/react";
import Title from "./Title";

describe("<Title />", () => {
  test("no data?", () => {
    render(<Title category="" />);

    const title = screen.queryByTestId("skillsTitle");
    expect(title).not.toBeInTheDocument();
    expect(title).toMatchSnapshot();
  });

  test("displays title ðŸ‘Œ", () => {
    render(<Title category="Monster Hunting" />);

    const title = screen.queryByTestId("skillsTitle");

    expect(title).toBeInTheDocument();
    expect(title).toMatchSnapshot();
  });
});
