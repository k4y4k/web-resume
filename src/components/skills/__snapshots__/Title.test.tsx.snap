// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> displays title ðŸ‘Œ 1`] = `
<h2
  data-testid="skillsTitle"
>
  Monster Hunting
</h2>
`;

exports[`<Title /> no data? 1`] = `null`;
